robots/cmd/retester/main.go

```diff
@@ -0,0 +1,308 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ * Copyright 2017 The Kubernetes Authors.
+ *
+ */
+
+// Derived from kubernetes/test-infra robots/commenter, since we need to improve
+// on filtering commit status:failure to only comment on required test lanes
+// failing
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sigs.k8s.io/prow/pkg/config"
+	"strconv"
+	"strings"
+	"time"
+
+	"sigs.k8s.io/prow/pkg/config/secret"
+	"sigs.k8s.io/prow/pkg/flagutil"
+	"sigs.k8s.io/prow/pkg/github"
+)
+
+const (
+	baseQuery = `
+		is:pr
+		archived:false
+		is:open
+		is:unlocked
+		-label:do-not-merge
+		-label:do-not-merge/blocked-paths
+		-label:do-not-merge/cherry-pick-not-approved
+		-label:do-not-merge/hold
+		-label:do-not-merge/invalid-owners-file
+		-label:do-not-merge/release-note-label-needed
+		-label:do-not-merge/work-in-progress
+		-label:needs-rebase
+		%s
+		status:failure
+		%s
+`
+)
+
+var (
+	repos = []string{
+		"kubevirt",
+		"kubevirtci",
+		"project-infra",
+	}
+
+	labelSets = []string{
+		"label:lgtm label:approved",
+		"label:skip-review",
+	}
+
+	fullQueries []string
+
+	comment = `/retest-required
+This bot automatically retries required jobs that failed/flaked on 
+required test lanes of PRs.
+Silence the bot with an ` + "`" + `/lgtm cancel` + "`" + ` or ` + "`" + `/hold` + "`" + ` comment for consistent failures.`
+
+	jobContextMatcher = regexp.MustCompile(`.*/([^/]+)/[0-9]+$`)
+
+	presubmitRequiredMap = map[string]struct{}{}
+)
+
+func init() {
+	var repoQueries []string
+	for _, r := range repos {
+		repoQueries = append(repoQueries, fmt.Sprintf("repo:kubevirt/%s", r))
+	}
+	for _, labelSet := range labelSets {
+		fullQueries = append(fullQueries, fmt.Sprintf(baseQuery, labelSet, strings.Join(repoQueries, " ")))
+	}
+}
+
+func flagOptions() options {
+	o := options{
+		endpoint: flagutil.NewStrings(github.DefaultAPIEndpoint),
+	}
+	flag.DurationVar(&o.updated, "updated", 2*time.Hour, "Filter to issues unmodified for at least this long if set")
```

Comment:
```suggestion
	flag.DurationVar(&o.updated, "updated", 2*time.Hour, "Duration to filter for unmodified issues")
```

Reply:
The original message implicitly implies by saying "if set" that you can set this value to zero and skip the restriction, which is removed by the suggested change.

Reply:
Duration to filter for unmodified issues if set ?